src/libostree/ostree-mutable-tree.c

```diff
@@ -304,6 +349,150 @@ ostree_mutable_tree_ensure_parent_dirs (OstreeMutableTree  *self,
   return ret;
 }
 
+static gboolean
+apply_whiteouts (OstreeMutableTree *self,
+                 OstreeMutableTree *layer,
+                 GError **error)
+{
+  GHashTableIter iter;
+  gpointer key, value;
+
+  g_hash_table_iter_init (&iter, layer->files);
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      const char *file_name = key;
+
+      if (!g_str_has_prefix (file_name, OSTREE_WHITEOUT_PREFIX))
+        continue;
+
+      if (strcmp (file_name, OSTREE_WHITEOUT_OPAQUE) == 0)
+        {
+          if (!ostree_mutable_tree_remove_all_children (self, error))
+            return FALSE;
+        }
+      else
+        {
+          g_autoptr(GError) local_error = NULL;
+          const char *whiteout_file_name =
+            file_name + strlen (OSTREE_WHITEOUT_PREFIX);
+
+          if (!ostree_mutable_tree_remove_child (self, whiteout_file_name, &local_error) &&
+              !g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            {
+              g_propagate_error (error, g_steal_pointer (&local_error));
```

Comment:
What do you mean? It especially ignores the "not found" error and only reports if you have some other error. So, removing something from the base layers thats not there should work. Did it not?